packages/fx-core/src/plugins/resource/bot/v2/plugin.ts

```diff
@@ -162,37 +176,51 @@ export class TeamsBotV2Impl {
   }
 
   private getTemplates(ctx: Context, inputs: Inputs): CodeTemplateInfo[] {
-    const actRoles = this.resolveActRoles(ctx, inputs);
-    const triggers = this.resolveTriggers(inputs);
-    const hostType = this.resolveHostType(ctx);
     const lang = this.resolveProgrammingLanguage(ctx);
-
-    const scenarios = this.resolveScenarios(actRoles, triggers, hostType);
-
-    return scenarios.map((scenario) => {
-      return {
+    const templateInfos: CodeTemplateInfo[] = [];
+    const botGroupScenarios = new Set<string>();
+    const msgextGroupScenarios = new Set<string>();
+    const solutionSettings = ctx.projectSetting.solutionSettings as AzureSolutionSettings;
+    const capabilities = solutionSettings.capabilities;
+    capabilities.map((capability: string) => {
```

Comment:
Let's sync more details about this logic. For VSCode, if teams app contains both ME and Bot, there is only one template is scaffolded instead of two. Let's align on the business scenario first.